packages/zoltan2/src/directory/Zoltan2_Directory_Impl.hpp

```diff
@@ -1084,15 +1023,15 @@ int Zoltan2_Directory<gid_t,lid_t,user_t>::remove(
   // eliminating remove_local and just calling here but will keep for now to
   // keep symmetry with other modes.
   if(nrec > 0) {
-    // node_map.begin_erase(); // TODO Switch to Kokkos Map
+    node_map.begin_erase();
     for (int i = 0; i < nrec; i++)  {
       msg_t *ptr = reinterpret_cast<msg_t*>(&(rbuff[i*remove_msg_size]));
```

Comment:
Does this obey ANSI strict aliasing rules?  (Remember the `memcpy` line of code I showed a little while ago?)

Reply:
This one doesn't give me the warning when I compile. It's a ptr cast versus the actual ptr write you showed me before.

I tried the following cases just to see how this works.
Only the 3rd case (matching the original memcpy case you suggested) gives the warning.

```
  // no strict-alias warning
  {
    char * char_ptr[10];
    int * ptr = reinterpret_cast<int *>(char_ptr);
  }

  // no strict-alias warning
  {
    char * char_ptr = new char[10];
    int * ptr = reinterpret_cast<int *>(char_ptr);
  }

  // strict-alias warning
  {
    char * char_ptr[10];
    int * ptr;
    *ptr = *(reinterpret_cast<int *>(char_ptr));
  }

  // no strict-alias warning
  {
    char * char_ptr = new char[10];
    int * ptr;
    *ptr = *(reinterpret_cast<int *>(char_ptr));
  }
```